### PR TITLE
mgr/dashboard: add helpers for compression in pool form

### PR DESCRIPTION
**Pull request by avanthakkar**
Fixes: https://tracker.ceph.com/issues/61297
Signed-off-by: avanthakkar <avanjohn@gmail.com>

Adding helpers for compression mode, algorithm, min/max blob size and compression ratio which is set to 0.875 as default. Also reduced size of pool name column. 
Screenshot from 2024-05-27 15-42-53

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.html [hunk 1]
```diff
@@ -421,9 +421,13 @@
...
-            <label i18n
-                   class="cd-col-form-label"
-                   for="mode">Mode</label>
+            <label class="cd-col-form-label"
+                   for="mode">
+              <ng-container i18n>Mode</ng-container>
+              <cd-helper>
+                <span i18n>Policy used for compression algorithm</span>
+              </cd-helper>
+            </label>
```
Comment (Pegonzal) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.html:430]: Are the `<ng-container>` and `<span>` tags even needed here?

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.html [hunk 2]
```diff
@@ -439,9 +443,13 @@
...
             <!-- Compression algorithm selection -->
             <div class="form-group row">
-              <label i18n
-                     class="cd-col-form-label"
-                     for="algorithm">Algorithm</label>
+              <label class="cd-col-form-label"
+                     for="algorithm">
+                <ng-container i18n>Algorithm</ng-container>
+                <cd-helper>
+                  <span i18n>Compression algorithm used for BlueStore</span>
```
Comment (Pegonzal) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.html:450]:
```suggestion
                  <span i18n>Compression algorithm used </span>
```
I don't think the user should know about BlueStore

**Review by Pegonzal — APPROVED** [review]
Overall lgtm!

Linting issue

`
[lint:html    ] 
[lint:html    ] [htmllint] found 0 errors out of 189 files
[lint:html    ] 
[lint:html    ] ERROR in /home/jenkins-build/build/workspace/ceph-pull-requests/src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.html
[lint:html    ] 
[lint:html    ] src/app/ceph/pool/pool-form/pool-form.component.html
[lint:html    ]   480:14  File should use 2 space indentation
[lint:html    ] 
[lint:html    ] 
[lint:eslint  ] 
[lint:eslint  ] Linting "ceph-dashboard"...
[lint:eslint  ] 
[lint:eslint  ] All files pass linting.
[lint:eslint  ] `

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.html (hunk 2 above)
Comment (cloudbehl) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.html:450]: I agree, Lets not use any non generic terms to explain to user.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.html [hunk 3]
```diff
@@ -463,9 +471,13 @@
...
             <!-- Compression min blob size -->
             <div class="form-group row">
-              <label i18n
-                     class="cd-col-form-label"
-                     for="minBlobSize">Minimum blob size</label>
+              <label class="cd-col-form-label"
+                     for="minBlobSize">
+                <ng-container i18n>Minimum blob size</ng-container>
+                <cd-helper>
+                  <span i18n>Chunks smaller than this are never compressed</span>
```
Comment (cloudbehl) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.html:478]:
```suggestion
                  <span i18n>Chunks smaller than this are never compressed</span>
```
```suggestion
                  <span i18n>Chunks smaller than Minimum blob size are never compressed</span>
```

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.html [hunk 4]
```diff
@@ -488,9 +500,13 @@
...
             <!-- Compression max blob size -->
             <div class="form-group row">
-              <label i18n
-                     class="cd-col-form-label"
-                     for="maxBlobSize">Maximum blob size</label>
+              <label class="cd-col-form-label"
+                     for="maxBlobSize">
+                <ng-container i18n>Maximum blob size</ng-container>
+                <cd-helper>
+                  <span i18n>Chunks larger than this are broken into smaller blobs of size mentioned before being compressed.</span>
```
Comment (cloudbehl) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.html:507]:
```suggestion
                  <span i18n>Chunks larger than `Maximum Blob Size` are broken into smaller blobs of size mentioned before being compressed.</span>
```

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.html [hunk 5]
```diff
@@ -512,9 +528,13 @@
...
             <!-- Compression ratio -->
             <div class="form-group row">
-              <label i18n
-                     class="cd-col-form-label"
-                     for="ratio">Ratio</label>
+              <label class="cd-col-form-label"
+                     for="ratio">
+                <ng-container i18n>Ratio</ng-container>
+                <cd-helper>
+                  <span i18n>The ratio of the size of the data chunk after compression relative to the original size must be at least this small in order to store the compressed version</span>
```
Comment (cloudbehl) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.html:535]: Still complicated. 
```suggestion
                  <span i18n>The ratio of the size of the data chunk after compression relative to the original size must be at least this small in order to store the compressed version</span>
```

**Review by pereman2 — APPROVED** [review]
Agree with comments but lgtm

**Comment by cloudbehl** [comment]
jenkins test make check

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.html (hunk 1 above)
Comment (nizamial09) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.html:430]: Yeah, that could be probably removed.
```suggestion
            <label class="cd-col-form-label"
                   for="mode"
                   i18n>Mode
              <cd-helper i18n>
                   Policy used for compression algorithm
              </cd-helper>
            </label>
```

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.ts [hunk 6]
```diff
@@ -128,7 +128,7 @@ export class PoolFormComponent extends CdForm implements OnInit {
       maxBlobSize: new FormControl('', {
         updateOn: 'blur'
       }),
-      ratio: new FormControl('', {
+      ratio: new FormControl('0.875', {
```
Comment (nizamial09) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.ts:131]: If this is a constant default value, lets initialize it at the start or outside the class as
`const DEFAULT_RATIO = 0.875` and use the constant here!

Comment (avanthakkar) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.ts:131]: Done!

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.html (hunk 5 above)
Comment (avanthakkar) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.html]: Thanks, updated the description!

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.html (hunk 1 above)
Comment (avanthakkar) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.html:430]: Fixed it, thanks!

**Comment by avanthakkar** [comment]
jenkins test make check

**Comment by avanthakkar** [comment]
jenkins test api

**Comment by avanthakkar** [comment]
jenkins test docs

**Comment by avanthakkar** [comment]
jenkins retest this please

**Review by afreen23 — CHANGES_REQUESTED** [review]
hey @avanthakkar , we are now using `<cd-help-text>` for help texts for forms.
You can look for an example in [bucket form page](https://github.com/ceph/ceph/blob/ef9bf7651800632d10bacef3c41f3432db769f40/src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-form/rgw-bucket-form.component.html#L202).

Thanks!

**Comment by avanthakkar** [comment]
> hey @avanthakkar , we are now using `<cd-help-text>` for help texts for forms. You can look for an example in [bucket form page](https://github.com/ceph/ceph/blob/ef9bf7651800632d10bacef3c41f3432db769f40/src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-form/rgw-bucket-form.component.html#L202).
> 
> Thanks!

Done! Also updated the screenshot

**Comment by avanthakkar** [comment]
jenkins test api

**Comment by avanthakkar** [comment]
jenkins test dashboard

**Comment by afreen23** [comment]
> > hey @avanthakkar , we are now using `<cd-help-text>` for help texts for forms. You can look for an example in [bucket form page](https://github.com/ceph/ceph/blob/ef9bf7651800632d10bacef3c41f3432db769f40/src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-form/rgw-bucket-form.component.html#L202).
> > Thanks!
> 
> Done! Also updated the screenshot

Thanks Avan!
We need the helper texts below the input field,  so you can perhaps take a look at the Buckets' create form.
image

https://github.com/ceph/ceph/blob/ef9bf7651800632d10bacef3c41f3432db769f40/src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-form/rgw-bucket-form.component.html#L198-L204

**Comment by avanthakkar** [comment]
> > > hey @avanthakkar , we are now using `<cd-help-text>` for help texts for forms. You can look for an example in [bucket form page](https://github.com/ceph/ceph/blob/ef9bf7651800632d10bacef3c41f3432db769f40/src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-form/rgw-bucket-form.component.html#L202).
> > > Thanks!
> > 
> > 
> > Done! Also updated the screenshot
> 
> Thanks Avan! We need the helper texts below the input field, so you can perhaps take a look at the Buckets' create form. image
> 
> https://github.com/ceph/ceph/blob/ef9bf7651800632d10bacef3c41f3432db769f40/src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-form/rgw-bucket-form.component.html#L198-L204

Done, thanks!

**Comment by avanthakkar** [comment]
jenkins test api

**Comment by avanthakkar** [comment]
jenkins test dashboard

**Review by afreen23 — APPROVED** [review]

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.html [hunk 7]
```diff
@@ -566,6 +577,11 @@
                      placeholder="e.g., 10GiB"
                      defaultUnit="GiB"
                      cdDimlessBinary>
+              <cd-help-text>
+                <span i18n>Leave it blank or specify 0 to disable this quota.</span>
+                <br>
+                <span i18n>A valid quota should be greater than 0.</span>
+              </cd-help-text>
```
Comment (afreen23) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.html:584]: Looks nice, just one thing, we  are using same help text for bytes and quota.
Is it by miss ? 

For quota:
Also its seem understood that 0 value will disable quota, means no quota, so perhaps you can rethink , if you want that to mention to user.
Rather we should specifiy what is meant by "max objects" and "max bytes" what will they do if user enters any value there ?

**Review by afreen23 — COMMENTED** [review]
Just one more minor thing I noticed, thanks!

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.html (hunk 7 above)
Comment (avanthakkar) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.html:584]: No, both apply to the quota. If specified as `0`, it means the quota is disabled. Given that they are part of quotas, it should be self-explanatory what the maximum bytes and maximum objects a user wants to consume.

**Comment by cloudbehl** [comment]
jenkins test api

**Comment by avanthakkar** [comment]
jenkins test api

**Comment by avanthakkar** [comment]
jenkins test dashboard cephadm

**Comment by avanthakkar** [comment]
jenkins test api

**Comment by cloudbehl** [comment]
@avanthakkar lets backport!

**Pull request merged**